### PR TITLE
Model vehicle traile

### PR DESCRIPTION
**Pull request by SaraML**
In commit 579c1f25638e04ff5d70aea503e55ef30a7928ed you can find the Simulink's simulation of vehicle-trailer. You can change the maneuver.
In commit 2948e1b619c05cbd0753b8dcba774e6aa1fc5de8 you can find the script for plotting the position of vehicle, hitch and trailer, provided by @raul-rd

### CODE REVIEW

**Review by jcontrolresearch — CHANGES_REQUESTED** [review]
On file: Source/MIL/Vehicle_trailer.slx
image
split bus in steering angle and parameter signals (vehicle_velocity, l1, l2, l12) and keep all outputs as single 

if preferr call

- L1: vehicle_axle_distance

- L12: Read_2_hitch_distance

- L2: Trailer_axle_distance

**Review by jcontrolresearch — CHANGES_REQUESTED** [review]
Including jackknife restrictions:

What are jackknife events?

Consider a trailer with single axle (picture just to figure out how's events looks like), vehicle is driving in reverse slowly (<10kmh) and no slip on tires

image

Ideally, in figure (b) trailer could move around [180 deg] range around the hitch point (joint to vehicle). In real life this condition is not possible due possible collision with trailer and vehicle as shows in figure (a). This conditions are considered not safety for drive assistance systems. The solution as shows in picture (c), is keep trailer restricted in range of angle movement from hitch. This kind of events are directly dependent from trailer wheelbase ,length and as trailer wide too.

The common variation on this parameters are described (ft) by:

image

image

Based on this data,  the geometrically jackknife angle could be calculated as:

image

Following this brief explanation:
TODO:

@SaraML Set a value for g parameter (based on common trailers)
@lfc-256  find the worst case scenario for width and length trailer variation from tables above
@raul-rd  include a function to limit JKK angle in simulation. The conditions is that is a jackknife occurred, vehicle and trailer should stop moving.

**Comment by SaraML** [comment]
There are 3 main styles of tongue
- The straight tongue
- The Compound tongue
- The "A" frame tongue

The longer a tongue can be (to a point where the tongue weight becomes excessive) the more smoothly and stable the trailer will travel. It will also be more responsive and forgiving when reversing.
Short tongue trailers are prone to rapid and sometimes unexpected turning, slight movements of the towing vehicle can put the trailer into a jackknife in a moment of inattention.
Most trailer manufacturers have their own standard length for a given size of trailer based on experience, trailer stability, turning and reversing responsiveness, axle position and the trailers maximum load capacity.
Trailer tongues need to be at least as long as half the tow vehicles width. Most cars and SUVs’ range between 5' (60" or 1.52m) to 6 1/2' (78" or 1.98m) so the minimum length of tongue needs to be half of this plus a little bit more (say 8-12" or 0.203-0.304m) to gain a margin of clearance when turning and to add stability to the trailer.
Many designers make tongues at 30" to 36" (0.76 to 0.91m), but it seems to be the sweet spot when the tongue gets to the 40" to 48" length (1.01 to 1.21m).

In summary, we can set the trailer tongue length (g) between **30" to 48"** (0.76 to 1.21m).

trailer3

trailer2

trailer1

**Comment by lfc-256** [comment]
In the next **table** we can see the relation of the tongue length  and width of trailer, finding the geometrically jackknife angle (in degrees).
Using the equation:

image


The variation of tongue length trailer increases in 1'', from 30'' to 48''.  

The maximum and minimum jackknife angle is highlighted in yellow.

We can find the worst case scenario for width and tongue length trailer variation.

image

**Comment by jcontrolresearch** [comment]
@SaraML , @lfc-256 great job, just update this information to section **Model validation wiki**

**Comment by raul-rd** [comment]
Update model to add  function to limit JKK angle in simulation. Please check commit 9292dd6bd5758fdd069a26be2a877482f7c3e7f6.
Update script for the model. Now show JKK angle and hitch angle. Please check commit 6d7bd277240bd86e4efb92e191e18a5d9bbb6297
